### ISSUE TITLE
Command Line Arguments

### ISSUE THREAD

**Issue by ahmedhassanme** [benjamin-hodgson/Contexts#4]
Hello!

It would be amazing if we were able to pass in command line arguments to 'given' functions eg

`run-contexts sometest.py -x arg1 -y arg2 -z arg3`

Contexts is awesome.

**Comment by benjamin-hodgson** [comment]
This is do-able; the framework has to mutate `sys.argv` after it's parsed its own arguments (and probably put it back again after the tests have finished). However we need a way of separating arguments you intended to pass to `run-contexts` from arguments you intended to pass to the tests themselves, in case they use the same flags.

Would something like this work for you?

```bash
$ run-contexts -s -v --teamcity --argv="-x arg1 -y arg2 -z arg3"
```

In this example `-s -v --teamcity` would be arguments for `run-contexts` itself; the things inside `--argv` would be passed through to the tests. Let me know what you think :wink:

**Comment by benjamin-hodgson** [comment]
Fixed in ff28df2
